ecosystem/typescript/sdk/src/tests/unit/type_tag.test.ts

```diff
@@ -81,4 +88,52 @@ describe("TypeTagParser", () => {
       expect(nestedTypeTag.value.type_args.length).toEqual(0);
     }
   });
+
+  describe("parse Object type", () => {
+    test("TypeTagParser successfully parses an Object type", () => {
+      const typeTag = "0x1::object::Object<T>";
+      const parser = new TypeTagParser(typeTag);
+      const result = parser.parseTypeTag();
+      expect(result instanceof TypeTagAddress).toBeTruthy();
+    });
+
+    test("TypeTagParser successfully parses a strcut with a nested Object type", () => {
+      const typeTag = "0x1::some_module::SomeResource<0x1::object::Object<T>>";
```

Comment:
For token v2, is this a valid tag? -> `0x1::some_module::SomeResource<0x1::object::Object<T>, 0x1::object::Object<T>>` Wonder if we can safeguard this scenario as well